ExampleCode/C++/API/API.cpp

```diff
@@ -0,0 +1,916 @@
+#include <iostream>
+#include <string>                    // std::string, std::to_string;
+#include "include/API.h"
+
+// To avoid poluting the namespace, and also to avoid typing std:: everywhere.
+using std::cin;
+using std::cout;
+using std::endl;
+using std::map;
+using std::string;
+using std::to_string;
+using std::vector;
+
+
+// Default constructor
+iSENSE::iSENSE()
+{
+  // Set these to default values for future reference
+  upload_URL = "URL";
+  get_URL = "URL";
+  get_UserURL = "URL";
+  title = "title";
+  project_ID = "empty";
+  contributor_key = "key";
+  contributor_label = "label";
+  email = "email";
+  password = "password";
+}
+
+
+// Constructor with parameters
+iSENSE::iSENSE(string proj_ID, string proj_title, string label, string contr_key)
+{
+  set_project_ID(proj_ID);
+  set_project_title(proj_title);
+  set_project_label(label);
+  set_contributor_key(contr_key);
+}
+
+
+// Similar to the constructor with parameters, but can be called at anytime to set up the upload object.
+void iSENSE::set_project_all(string proj_ID, string proj_title, string label, string contr_key)
+{
+  set_project_ID(proj_ID);
+  set_project_title(proj_title);
+  set_project_label(label);
+  set_contributor_key(contr_key);
+}
+
+
+// This function should be called by the user, and should set up all the fields.
+void iSENSE::set_project_ID(string proj_ID)
+{
+  // Set the Project ID, and the upload/get URLs as well.
+  project_ID = proj_ID;
+  upload_URL = devURL + "/projects/" + project_ID + "/jsonDataUpload";
+  get_URL = devURL + "/projects/" + project_ID ;
+  get_project_fields();
+}
+
+
+// The user should also set the project title
+void iSENSE::set_project_title(string proj_title)
+{
+  title = proj_title;
+}
+
+
+// This one is optional, by default the label will be "cURL".
+void iSENSE::set_project_label(string proj_label)
+{
+  contributor_label = proj_label;
+}
+
+
+// As well as the contributor key they will be using
+void iSENSE::set_contributor_key(string proj_key)
+{
+  contributor_key = proj_key;
+}
+
+
+// Sets the email address to be used to upload data
+void iSENSE::set_email(string proj_email)
+{
+  email = proj_email;
+}
+
+
+// Sets the password to be used
+void iSENSE::set_password(string proj_password)
+{
+  password = proj_password;
+}
+
+
+// Extra function that the user can call to just generate a timestamp
+// IT DOES NOT push back to the map of vectors! It merely returns a string,
+// that users may grab and then send off to the push_back function.
+string iSENSE::generate_timestamp(void)
+{
+  // Generates a ISO 8601 timestamp
+  time_t time_stamp;
+  time(&time_stamp);
+  char buffer[sizeof "2011-10-08T07:07:09Z"];
+
+  // This originally did not compile in Visual Studios.
+  // Note that "%F" and "%T" are not supported by Microsoft.
+  strftime(buffer, sizeof buffer, "%Y-%m-%dT%H:%M:%SZ", gmtime(&time_stamp));

```

Comment:
Originally I used something like:
strftime(buf, sizeof buf, "%FT%TZ", gmtime(&now));
But it turns out some characters aren't supported by Microsoft, so I switched it to use all the supported ones.
It does currently compile so I'll just change that comment to something relevant.